mongoengine/fields.py

```diff
@@ -104,8 +105,24 @@ def __init__(self, regex=None, max_length=None, min_length=None, **kwargs):
         self.regex = re.compile(regex) if regex else None
         self.max_length = max_length
         self.min_length = min_length
+        self._kwargs = kwargs
         super().__init__(**kwargs)
 
+    def __deepcopy__(self, memo):
+        memo[id(self)] = self
```

Comment:
I believe this is already done in the call `copied_obj = copy.deepcopy(self, memo)` a few line below

Reply:
Yes...I found I got it wrong after some digging.
`memo[id(self)] = self` actually make `deepcopy(self)` bypass deepcopy operate totally.